modules/node/node.module

```diff
@@ -746,7 +746,7 @@ function node_load($param = array(), $revision = NULL, $reset = NULL) {
  *   A fully-populated node object.
  */
 function _node_load_direct($param = array(), $revision = NULL, $reset = NULL) {
-  static $nodes = array();
+  $nodes = &drupal_static('node_load', array());
 
   if ($reset) {
```

Comment:
Hmm so it does by setting $nodes = array(). Since we're in the same function that's fine. So yes looks great.